src/NUnitFramework/framework/Internal/Commands/SetUpTearDownItem.cs

```diff
@@ -92,11 +92,35 @@ private void RunSetUpOrTearDownMethod(TestExecutionContext context, IMethodInfo
             _methodValidator?.Validate(method.MethodInfo);
 
             var methodInfo = MethodInfoCache.Get(method);
+            RaiseStartedEvent(context, methodInfo);
 
-            if (methodInfo.IsAsyncOperation)
-                AsyncToSyncAdapter.Await(() => InvokeMethod(method, context));
-            else
-                InvokeMethod(method, context);
+            try
+            {
+                if (methodInfo.IsAsyncOperation)
+                    AsyncToSyncAdapter.Await(() => InvokeMethod(method, context));
+                else
+                    InvokeMethod(method, context);
+            }
+            finally
+            {
+                RaiseFinishedEvent(context, methodInfo);
+            }
+        }
+
+        private static void RaiseStartedEvent(TestExecutionContext context, MethodInfoCache.TestMethodMetadata methodInfo)
+        {
+            if (methodInfo.HasOneTimeSetUpAttribute)
+                context.ListenerExt.OneTimeSetUpStarted(context.CurrentTest);
+            else if (methodInfo.HasOneTimeTearDownAttribute)
+                context.ListenerExt.OneTimeTearDownStarted(context.CurrentTest);
+        }
+
```

Comment:
Do you need this if inside here, you know when you call what you have, you can just pass that along, or even better drop this method all together and just call in in-place.

Reply:
Right. Updated in latest commit.